shared/AppInsightsCore/src/JavaScriptSDK/StatsBeat.ts

```diff
@@ -0,0 +1,182 @@
+                    _networkCounter.failure[status] = (_networkCounter.failure[status] || 0) + 1;
+                }
+                _setupTimer();
+            };
+
+            
+            
+            _self.countException = (endpoint: string, exceptionType: string) => {
+                if (!_isEnabled || !_checkEndpoint(endpoint)) {
+                    return;
+                }
+                _networkCounter.exception[exceptionType] = (_networkCounter.exception[exceptionType] || 0) + 1;
+                _setupTimer();
+            }
+
+            function _setupTimer() {
+                if (!_timeoutHandle) {
+                    _timeoutHandle = scheduleTimeout(() => {
+                        _timeoutHandle = null;
+                        this.trackShortIntervalStatsbeats();
+                    }, STATS_COLLECTION_SHORT_INTERVAL);
+                }
+            }
+
+            _self.trackShortIntervalStatsbeats = (): void => {
+                _trackSendRequestDuration();
+                _trackSendRequestsCount();
+                _networkCounter = createNetworkStatsbeat(_networkCounter.host);
+                _timeoutHandle && _timeoutHandle.cancel();
+                _timeoutHandle = null;
+            }
+
+            function _checkEndpoint(endpoint: string) {
+                return _networkCounter && _networkCounter.host === endpoint;
+            }
+
+            function _getCustomProperties(ikey: string) {
+                _cikey = ikey;
+                _language = STATSBEAT_LANGUAGE;
+                _os = STATSBEAT_TYPE;
+                _runTimeVersion = STATSBEAT_TYPE;
+            }
+
+            function _sendStatsbeats(name: string, val: number, properties?: {}) {
+                // Add extra properties
+                let baseProperties = {
+                    "rp": "unknown",
+                    "attach": "Manual",
+                    "cikey": _cikey,
+                    "runtimeVersion": _runTimeVersion,
+                    "os": _os,
+                    "language": _language,
+                    "version": _sdkVersion,
+                    "endpoint": "breeze",
+                    "host": _networkCounter.host
+                }
+                let statsbeatEvent: ITelemetryItem = {
+                    iKey: INSTRUMENTATION_KEY,
+                    name: name,
+                    baseData: {
+                        name: name,
+                        average: val,
+                        properties: {"host": _networkCounter.host, ...properties, ...baseProperties}
+                    },
+                    baseType: "MetricData"
+                };
+                _core.track(statsbeatEvent);
+            }
+
+            function _trackSendRequestDuration() {
+                var currentCounter = _networkCounter;
+                if (currentCounter.totalRequest > 0 ) {
+                    var averageRequestExecutionTime = currentCounter.requestDuration  / currentCounter.totalRequest;
```

Comment:
I don't think a cumulative duration across intervals is the duration we want. We want to get the average duration WITHIN an export interval, so you need to reset the duration and count after each export.